src/scripts/calc.js

```diff
@@ -1,95 +1,109 @@
 /*eslint-env node*/
-/*globals w2popup model*/
 
-var $ = require('jquery');
-var model = require('./model');
+var ref = require("./firebase_ref");
 
-var m=document.createElement('button');
-    m.className = 'commands';
-    m.setAttribute('id', 'calculate');
-    m.innerHTML = 'Calculate';
-    document.body.appendChild(m);
-    
-function parseResponse(response){
+function converted(obj) {
+    var t = {};
+    Object.keys(obj).forEach(function(key) {
+        t[key] = Object.keys(obj[key]).map(function(k) {
+            var u = obj[key][k];
+            u.recid = parseInt(k, 10);
+            return u;
+        });
+    });
+    return t;
+}
+
+function parseResponse(response) {
     var r = [];
     var d = response.result.displacements;
-    for (p in d){
-        r.push({recid:p, x:d[p].x, y:d[p].y, z:d[p].z});
+    for (p in d) {
```

Comment:
__[jshint]__
* Creating global 'for' variable. Should be 'for (var p ...'.

<!-- Comment by [SideCI](https://sideci.com) -->